tests/test_views.py

```diff
@@ -631,3 +631,64 @@ def test_when_correct_otp_email_passed(self):
         response = self.client.post(self.url, data=self.data_correct_otp, format="json")
 
         self.assertEqual(202, response.status_code)
+
+
+class TestUploadImageView(APITestCase):
+    """UploadImageView Test"""
+
+    def setUp(self) -> None:
+        """SetUp test data"""
+        self.url = reverse("upload_profile_image")
+
+        self.user = baker.make(
+            "drf_user.User",
+            username="user",
+            email="user@email.com",
+            name="user",
+            mobile=1234569877,
+            is_active=True,
+        )
+
+        self.auth_transaction = baker.make(
+            "drf_user.AuthTransaction",
+            created_by=self.user,
+        )
+
+        # create otp of registered user
+        self.user_otp = baker.make(
+            "drf_user.OTPValidation", destination="user@email.com", otp=437474
+        )
+
+        self.user.set_password("pass123")
+        self.user.save()
+
+    @pytest.mark.django_db
+    def test_object_created(self):
+        """Check if the User object is created or not"""
+        self.assertEqual(1, User.objects.count())
+        self.assertEqual(1, AuthTransaction.objects.count())
+
+    @pytest.mark.django_db
+    def test_when_nothing_is_passed(self):
+        """Check when nothing is passed as data then api raises 400"""
+
+        self.client.force_authenticate(self.user)
+        response = self.client.post(self.url, data={}, format="multipart")
+
+        self.assertEqual(400, response.status_code)
+        self.assertEqual("No file was submitted.", response.json()["profile_image"][0])
+
+    @pytest.mark.django_db
+    def test_when_upload_image_passed(self):
+        """Check when correct otp and email is passed as data then api raises 201"""
+        import os
+
+        RES_DIR = os.path.join(os.path.abspath(os.path.dirname(__file__)), "res")
```

Comment:
@iamvivekkaushik Please don't use RES_DIR or something like this, please create a fixtures dir in tests directory. Then you can add a dummy image in fixtures dir for running tests.